src/imgtools/ops/ops.py

```
@@ -133,30 +135,43 @@ def __init__(self,
             logger.debug("Output exists, force updating.", path=self.csv_path)
             logger.info("Indexing the dataset...")
-            db = crawl(self.dir_path, n_jobs=n_jobs)
-            logger.info(f"Number of patients in the dataset: {len(db)}")
+            self.db = crawl(self.dir_path, n_jobs=n_jobs)
+            logger.info(f"Number of patients in the dataset: {len(self.db)}")
         else:
+            logger.info("Output exists, loading existing index.")
+            if not self.json_path.exists():
+                raise FileNotFoundError(f"Index file {self.json_path} not found, please re-index the dataset.")
+
+            with json_path.open("r") as f:
```

Ruff (0.8.2)

Undefined name `json_path`

(F821)